### ISSUE TITLE
cli

### ISSUE THREAD

**Issue by suntong** [mh-cbon/emd#3]
Thanks a lot for your fast fixes. 

This is only a FYI -- I noticed that you are writing your own cli package, so I'm just letting you know that there is an exiting cli package that supports all your current needs and provides more functionalities as well. Take a look at https://github.com/suntong/lang/tree/master/lang/Go/src/sys/CLI, the sample file _name_ give pretty clear picture of what the demo is about, and the source code is one-click away. 

Using it, 

- you will have the advantages of all your current need, and future needs as well. I especially like its input/output file handling (ref: https://github.com/suntong/cascadia/blob/master/cascadia.yaml), and many more. 
- moreover, you can **_declare_** your cli in Yaml format, 
https://github.com/mkideal/cli/wiki/Jump-start-the-cli-based-project#source
and the rest can pretty much all be taken care of. 

just my 2c.

**Comment by mh-cbon** [comment]
yeah, you have right, i m re inventing the wheel here.
Previously i used urfave/cli, but not this time, 
i was influenced by some project read on github based on std flags.

After this experiment i m still not very satisfied by std flag package, 
urfave/cli is a bit heavy to write : s
I feel like mkideal/cli is similar to urfave/cli.

Let s see : )

**Comment by suntong** [comment]
Just FYI, so closing it...

**Comment by mh-cbon** [comment]
might be fixed later.

**Comment by suntong** [comment]
Yeah, good idea. If you take a look at https://github.com/suntong/cascadia again, you'll find that I've added another parameter, `-p, --piece`, which basically accepts (_multiple_) arguments like `x=abc`. 

I.e., the `cli` package is extremely flexible and powerful. and you can see that it can even support argument format like `x=MYTAG:abc`, or whatever necessary for you. And still, the boilerplate can still be automatically generated, AND regenerated in a round-robin way.
